client/ayon_ftrack/plugins/publish/collect_webpublisher_credentials.py

```diff
@@ -91,13 +93,19 @@ def _get_username_key(self, context):
         api_key_secret = service_settings["api_key"]
         username_secret = service_settings["username"]
 
-        secrets_by_name = {
-            secret["name"]: secret["value"]
-            for secret in ayon_api.get_secrets()
-        }
+        con = ayon_api.get_server_api_connection()
+        with con.as_username(self.service_user_name):
```

Comment:
Wait, what? Why?

Reply:
We need to query secrets to get ftrack credentials, so service account should be used only for that.